dotnet/src/Connectors/Connectors.Memory.Postgres/PostgresCollectionOptions.cs

```diff
@@ -29,4 +29,9 @@ public sealed class PostgresCollectionOptions
     /// Gets or sets the default embedding generator for vector properties in this collection.
     /// </summary>
     public IEmbeddingGenerator? EmbeddingGenerator { get; init; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the data source should be disposed after the collection is disposed.
+    /// </summary>
+    internal bool OwnsDataSource { get; set; }
```

Comment:
Please keep in mind it's internal as it's not set via the option bag by the user (it's set by the user in the store/collection ctor and or when we pass it around)